ambry-network/src/main/java/com.github.ambry.network/Selector.java

```diff
@@ -347,13 +356,32 @@ public void poll(long timeoutMs, List<NetworkSend> sends)
           close(key);
         }
       }
+      completeSslHandshakes();
       this.metrics.selectorIORate.inc();
     }
     long endIo = time.milliseconds();
     this.metrics.selectorIOTime.update(endIo - endSelect);
   }
 
   /**
+   * Add Ssl connections to connected list on handshake completion
+   */
+  private void completeSslHandshakes() {
```

Comment:
This method does not do what its name implies, could you rename it to `checkHandshakeStatus()` or something?